### PR TITLE
BaSP-MR-27: test activities get and post

### PR DESCRIPTION
**Pull request by Micaela-Rossi**
Added test for get, get by id and post.
One test for GET it's at the end of the file because to test it, had to clean the seed, and if you let the test up, the other test fail, because are no data in the seed.

### CODE REVIEW

File: src/tests/activity.test.js [hunk 1]
```diff
@@ -1 +1,141 @@
...
+beforeAll(async () => {
+  await activity.collection.insertMany(activitySeed);
+});
+
+describe('GET /api/activities', () => {
+  test('The database must return 200 if is not empty of activities', async () => {
+    const response = await request(app).get('/api/activities').send();
+    expect(response.body.data.length).toBeGreaterThan(0);
+    expect(response.status).toBe(200);
+    expect(response.error).toBeFalsy();
```
Comment (Micaela-Rossi) [inline src/tests/activity.test.js:45]: I just watch the review and now i'm gonna change it to response.body.error

File: src/tests/activity.test.js [hunk 2]
```diff
@@ -1 +1,141 @@
...
+  test('If are activities, data must said "Complete activities list" ', async () => {
+    const response = await request(app).get('/api/activities').send();
+    expect(response.body.message).toMatch(/Complete activities list/);
+    expect(response.error).toBeFalsy();
+  });
+});
+
+describe('GET BY ID /api/activities/:id', () => {
+  test('The database must return 200 if the id exist', async () => {
+    const response = await request(app).get('/api/activities/64677fdefc13ae39f1753b24').send();
```
Comment (Micaela-Rossi) [inline src/tests/activity.test.js:61]: I will change this to a constant that save the id from seed, so is not manual putting the number

File: src/tests/activity.test.js [hunk 3]
```diff
@@ -1 +1,141 @@
...
+
+beforeAll(async () => {
+  await activity.collection.insertMany(activitySeed);
+});
+
+describe('GET /api/activities', () => {
+  test('The database must return 200 if is not empty of activities', async () => {
+    const response = await request(app).get('/api/activities').send();
+    expect(response.body.data.length).toBeGreaterThan(0);
+    expect(response.status).toBe(200);
```
Comment (nailaghigo) [inline src/tests/activity.test.js:44]: We need to test also `response.message`

Also test this throughout all the file.

File: src/tests/activity.test.js [hunk 4]
```diff
@@ -1 +1,141 @@
...
+    expect(response.error).toBeFalsy();
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/activities').send();
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
```
Comment (nailaghigo) [inline src/tests/activity.test.js:50]: We should access response.body.error
```suggestion
    expect(response.body.error).toBeTruthy();
```

Please apply this through all the file.

File: src/tests/activity.test.js (hunk 2 above)
Comment (nailaghigo) [inline src/tests/activity.test.js:61]: What happens if we change the admin's ID in the seed file? I think we should use a variable for the ID and we should capture the ID of the first admin of the seed.

Consider this every time we use admin ID

Comment (Micaela-Rossi) [reply src/tests/activity.test.js:61]: Here i try to use a function to get the first id from seed, but the id is called _id, and eslint doesn't allow using a '_'. How can i solve it?


My code is:
`let firstActivityId;

beforeAll(async () => {
  await activity.collection.insertMany(activitySeed);
  const foundFirstId = await activity.find();
  if (foundFirstId.length > 0) {
    firstActivityId = foundFirstId[0]._id;
  }
});`

File: src/tests/activity.test.js [hunk 5]
```diff
@@ -1 +1,141 @@
...
+  isActive: true,
+};
+
+const mockLessIsActive = {
+  name: 'Names',
+  description: 'some description',
+  isActive: '',
+};
+
+beforeAll(async () => {
```
Comment (nailaghigo) [inline src/tests/activity.test.js:36]: Maybe we should try this Instead of using beforeAll(), to avoid lines 132 to 140
```
beforeEach(async () => {
  await Activity.collection.insertMany(activitySeed);
});

afterEach(async () => {
  await Activity.collection.deleteMany();
});
```

**Review by nailaghigo — CHANGES_REQUESTED** [review]
💯

File: src/tests/activity.test.js (hunk 2 above)
Comment (Micaela-Rossi) [reply src/tests/activity.test.js:61]: I foun that i can use // eslint-disable-next-line no-underscore-dangle  but i'm not sure if it's okay or maybe doing all the logic in other way

File: src/tests/activity.test.js (hunk 4 above)
Comment (Micaela-Rossi) [reply src/tests/activity.test.js:50]: doesn't pass the tests if i apply this changes

File: src/tests/activity.test.js (hunk 2 above)
Comment (nailaghigo) [reply src/tests/activity.test.js:61]: It's ok to use that ESLint rule.

File: src/tests/activity.test.js (hunk 5 above)
Comment (Micaela-Rossi) [reply src/tests/activity.test.js:36]: fixed

File: src/tests/activity.test.js (hunk 4 above)
Comment (Micaela-Rossi) [reply src/tests/activity.test.js:63]: In error 404 changed to toBeUndefined()

File: src/tests/activity.test.js [hunk 6]
```diff
@@ -1 +1,149 @@
...
+    expect(response.body.error).toBeFalsy();
+    expect(response.body).toBeDefined();
+    expect(response.body.message).toMatch('Complete activities list');
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/activities').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
+  });
+  describe('GET /api/activities', () => {
```
Comment (Dannylez) [inline src/tests/activity.test.js:65]: Just out of curiosity, why did you add a new "describe" if it's the same as the previous one?

Comment (Micaela-Rossi) [reply src/tests/activity.test.js:65]: I forgot to delete it, before it was at the end of the file so i needeed to describe again but i forgot to delete it when i corrected with the before each

File: src/tests/activity.test.js [hunk 7]
```diff
@@ -1 +1,149 @@
...
+
+describe('GET BY ID /api/activities/:id', () => {
+  test('The database must return 200 if the id exist', async () => {
+    const response = await request(app).get(`/api/activities/${firstActivityId}`).send();
+    expect(response.status).toBe(200);
+    expect(response.body).toBeDefined();
+    expect(response.body.data).toHaveProperty('_id');
+    expect(response.body.data).toHaveProperty('name');
+    expect(response.body.data).toHaveProperty('description');
+    expect(response.body.data).toHaveProperty('isActive');
```
Comment (Dannylez) [inline src/tests/activity.test.js:84]: To save some lines of code, you could use a single .toHaveProperty() and list all the properties you want to evaluate within it, separated by commas. 
```suggestion
    expect(response.body.data).toHaveProperty('_id', 'name', 'description', 'isActive');
```

Comment (Micaela-Rossi) [reply src/tests/activity.test.js:84]: Great, i didn't know i can do this. Thanks!

**Review by Dannylez — APPROVED** [review]
🚀

**Review by nailaghigo — APPROVED** [review]
